tests/lib/rules/arrow-body-style.js

```diff
@@ -45,6 +46,191 @@ ruleTester.run("arrow-body-style", rule, {
+            code: "function foo(){ for (let a = (b, c, d) => { return v in b && c in d; }; ;); }",
+            output: "function foo(){ for (let a = (b, c, d) => (v in b && c in d); ;); }",
+            errors: [
+                {
+                    line: 1,
+                    column: 43,
+                    messageId: "unexpectedSingleBlock"
+                }
+            ]
+        },
+        {
+            code: "for ( a = (b, c, d) => { return v in b && c in d; }; ;);",
+            output: "for ( a = (b, c, d) => (v in b && c in d); ;);",
+            errors: [
+                {
+                    line: 1,
+                    column: 24,
+                    messageId: "unexpectedSingleBlock"
+                }
+            ]
+        },
+        {
+            code: "for ( a = (b) => { return (c in d) }; ;);",
+            output: "for ( a = (b) => (c in d); ;);",
+            errors: [
+                {
+                    line: 1,
+                    column: 18,
+                    messageId: "unexpectedSingleBlock"
+                }
+            ]
+        },
+        {
+            code: "for (let a = (b, c, d) => { return vb in dd ; }; ;);",
+            output: "for (let a = (b, c, d) => (vb in dd ); ;);",
+            errors: [
+                {
+                    line: 1,
+                    column: 27,
+                    messageId: "unexpectedSingleBlock"
+                }
+            ]
+        },
+        {
+            code: "for (let a = (b, c, d) => { return vb in c in dd ; }; ;);",
+            output: "for (let a = (b, c, d) => (vb in c in dd ); ;);",
+            errors: [
+                {
+                    line: 1,
+                    column: 27,
+                    messageId: "unexpectedSingleBlock"
+                }
+            ]
+        },
+        {
+            code: "do{let a = () => {return f in ff}}while(true){}",
+            output: "do{let a = () => (f in ff)}while(true){}",
+            errors: [{
+                line: 1,
+                column: 18,
+                messageId: "unexpectedSingleBlock"
+            }]
+        },
+        {
+            code: "do{for (let a = (b, c, d) => { return vb in c in dd ; }; ;);}while(true){}",
+            output: "do{for (let a = (b, c, d) => (vb in c in dd ); ;);}while(true){}",
+            errors: [{
+                line: 1,
+                column: 30,
+                messageId: "unexpectedSingleBlock"
+            }]
+        },
+        {
+            code: "scores.map(score => { return x in +(score / maxScore).toFixed(2)});",
+            output: "scores.map(score => (x in +(score / maxScore).toFixed(2)));",
+            errors: [{
+                line: 1,
+                column: 21,
+                messageId: "unexpectedSingleBlock"
+            }]
+        },
+        {
+            code: "const fn = (a, b) => { return a + x in Number(b) };",
+            output: "const fn = (a, b) => (a + x in Number(b));",
```

Comment:
We should check whether or not the arrow function is in a for-loop initializer, because that's the only case that causes problems with the `in` keyword (it makes the loop being interpreted as a `for-in` loop).

Maybe a function `IsInsideForLoopInitializer(node)` which returns `true` or `false`.

Reply:
done